### PR TITLE
Fix 3 lgtm warnings

### PR DESCRIPTION
**Pull request by 08A**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**
1- fix possible overflow
2- fix possible overflow
3- fix the usage of asctime

**Test plan**
Run the test suite and watch lgtm result

**Closing issues**
Work #16493

### CODE REVIEW

File: libr/bin/pdb/pdb.c [hunk 1]
```diff
@@ -485,7 +485,7 @@ static bool pdb7_parse(RPdb *pdb) {
 	}
 	p_tmp = root_page_data;
 	for (i = 0; i < num_root_index_pages; i++) {
-		r_buf_seek (pdb->buf, root_index_pages[i] * page_size,
+		r_buf_seek (pdb->buf, (st64)root_index_pages[i] * (st64)page_size,
```
Comment (trufae) [inline libr/bin/pdb/pdb.c:488]: use the new OVF apis instead of this problematic signed computation

Comment (trufae) [reply libr/bin/pdb/pdb.c:488]: if (UT64_MUL_OVFCHK(root_index_pages[i], page_size)) { break; }

File: libr/core/cmd_print.c [hunk 2]
```diff
@@ -5417,7 +5418,7 @@ l = use_blocksize;
 						core->num->value = r_core_print_disasm (core->print,
 							core, addr, block1, addrbytes * l, l, 0, 1, formatted_json, NULL, NULL);
 					} else {
-						eprintf ("Cannot allocate %d byte(s)\n", addrbytes * l);
+						eprintf ("Cannot allocate %ld byte(s)\n", addrbytes * l);
```
Comment (trufae) [inline libr/core/cmd_print.c:5421]: it's %zd if you use size_t

File: libr/util/format.c [hunk 3]
```diff
@@ -506,7 +506,8 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
 	if (MUSTSET) {
 		p->cb_printf ("wv4 %s @ 0x%08"PFMT64x"\n", setval, seeki+((elem>=0)?elem*4:0));
 	} else if (MUSTSEE) {
-		char *timestr = strdup(asctime (gmtime_r ((time_t*)&addr, &timestruct)));
+		char *timestr = malloc(26 * sizeof(char));
```
Comment (trufae) [inline libr/util/format.c:509]: missing space before (. and sizeof(char) is always1, so there's no need to add such multiplication

File: libr/util/format.c [hunk 4]
```diff
@@ -506,7 +506,8 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
...
+		asctime_r (gmtime_r ((time_t*)&addr, &timestruct), timestr);
```
Comment (trufae) [inline libr/util/format.c:510]: cc @thestr4ng3r

File: libr/util/format.c [hunk 5]
```diff
@@ -542,7 +542,8 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
 		}
 		free (timestr);
 	} else if (MUSTSEEJSON || MUSTSEESTRUCT) {
-		char *timestr = strdup (asctime (gmtime_r ((time_t*)&addr, &timestruct)));
+		char *timestr = malloc(26 * sizeof(char));
```
Comment (trufae) [inline libr/util/format.c:545]: same

**Comment by 08A** [comment]
@trufae In the radare2 repo is a safe malloc wrapper (such as xmalloc, ect ...)

File: libr/bin/pdb/pdb.c [hunk 6]
```diff
@@ -485,6 +485,7 @@ static bool pdb7_parse(RPdb *pdb) {
 	}
 	p_tmp = root_page_data;
 	for (i = 0; i < num_root_index_pages; i++) {
+		if (UT64_MUL_OVFCHK(root_index_pages[i], page_size)) { break; }
```
Comment (trufae) [inline libr/bin/pdb/pdb.c:488]: Follow the coding style guidelines:
* missing space before (
* newlines after { and ;

File: libr/util/format.c [hunk 7]
```diff
@@ -506,7 +506,8 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
 	if (MUSTSET) {
 		p->cb_printf ("wv4 %s @ 0x%08"PFMT64x"\n", setval, seeki+((elem>=0)?elem*4:0));
 	} else if (MUSTSEE) {
-		char *timestr = strdup(asctime (gmtime_r ((time_t*)&addr, &timestruct)));
+		char *timestr = malloc (26);
```
Comment (trufae) [inline libr/util/format.c:509]: malloc can fail

**Review by trufae — DISMISSED** [review]

File: libr/util/format.c (hunk 7 above)
Comment (08A) [reply libr/util/format.c:509]: @trufae Is there a safe malloc wrapper in radare2 (xmalloc, ect ...)

File: libr/bin/pdb/pdb.c (hunk 6 above)
Comment (08A) [reply libr/bin/pdb/pdb.c:488]: One problem with the usage of UT64_MUL_OVFCHK is that we still have the lgtm warning (Multiplication result converted to larger type).

Comment (08A) [reply libr/bin/pdb/pdb.c]: We could maybe use some comment to discard the warning.

Comment (08A) [reply libr/bin/pdb/pdb.c]: https://lgtm.com/help/lgtm/alert-suppression#c-c++

File: libr/core/cmd_print.c [hunk 8]
```diff
@@ -10,7 +10,7 @@
 #define R_CORE_MAX_DISASM (1024 * 1024 * 8)
 #define PF_USAGE_STR "pf[.k[.f[=v]]|[v]]|[n]|[0|cnt][fmt] [a0 a1 ...]"
 
-static int printzoomcallback(void *user, int mode, ut64 addr, ut8 *bufz, ut64 size);
+static int printzoomcallback (void *user, int mode, ut64 addr, ut8 *bufz, ut64 size);
```
Comment (ret2libc) [inline libr/core/cmd_print.c:13]: it was right as it was before.

File: libr/core/cmd_print.c [hunk 9]
```diff
@@ -443,27 +443,37 @@ static const char *help_msg_pi[] = {
 };
 
 static const char *help_msg_pif[] = {
-	"Usage:", "pif[cj]", "",
-	"pif?", "", "print this help message",
-	"pifc", "", "print all calls from this function",
-	"pifcj", "", "print all calls from this function in JSON format",
-	"pifj", "", "print instructions of function in JSON format",
+	"Usage:",
```
Comment (ret2libc) [inline libr/core/cmd_print.c:446]: no please, it is better to have the format there was before.

File: libr/core/cmd_print.c [hunk 10]
```diff
@@ -573,31 +583,271 @@ const char *help_msg_pxA[] = {
...
+	0xff0faf,
+	0xff28b7,
+	0xff43c0,
+	0xff5ec9,
+	0xff79d2,
+	0xffffff,
 };
 
-static void cmd_print_init(RCore *core, RCmdDesc *parent) {
+static void cmd_print_init (RCore *core, RCmdDesc *parent) {
```
Comment (ret2libc) [inline libr/core/cmd_print.c:844]: again, it was right without the space.

File: libr/core/cmd_print.c [hunk 11]
```diff
@@ -573,31 +583,271 @@ const char *help_msg_pxA[] = {
...
 	DEFINE_CMD_DESCRIPTOR_SPECIAL (core, &, amper);
 	DEFINE_CMD_DESCRIPTOR_SPECIAL (core, @, at);
-	DEFINE_CMD_DESCRIPTOR_SPECIAL (core, @@, at_at);
+	DEFINE_CMD_DESCRIPTOR_SPECIAL (core, @ @, at_at);
```
Comment (ret2libc) [inline libr/core/cmd_print.c:847]: wrong, revert this.

File: libr/core/cmd_print.c [hunk 12]
```diff
@@ -573,31 +583,271 @@ const char *help_msg_pxA[] = {
...
 	DEFINE_CMD_DESCRIPTOR (core, p);
-	DEFINE_CMD_DESCRIPTOR_SPECIAL (core, p=, p_equal);
-	DEFINE_CMD_DESCRIPTOR_SPECIAL (core, p-, p_minus);
+	DEFINE_CMD_DESCRIPTOR_SPECIAL (core, p =, p_equal);
```
Comment (ret2libc) [inline libr/core/cmd_print.c:849]: revert also this.

File: libr/core/cmd_print.c [hunk 13]
```diff
@@ -573,31 +583,271 @@ const char *help_msg_pxA[] = {
...
+	DEFINE_CMD_DESCRIPTOR_SPECIAL (core, p -, p_minus);
```
Comment (ret2libc) [inline libr/core/cmd_print.c:850]: revert also this.

File: libr/core/cmd_print.c [hunk 14]
```diff
@@ -609,7 +859,7 @@ static void cmd_print_init(RCore *core, RCmdDesc *parent) {
 	DEFINE_CMD_DESCRIPTOR (core, pz);
 }
 
-static void __cmd_pad(RCore *core, const char *arg) {
+static void __cmd_pad (RCore *core, const char *arg) {
```
Comment (ret2libc) [inline libr/core/cmd_print.c:862]: wrong

File: libr/core/cmd_print.c [hunk 15]
```diff
@@ -626,7 +876,7 @@ static void __cmd_pad(RCore *core, const char *arg) {
 }
 
 // colordump
-static void cmd_prc(RCore *core, const ut8* block, int len) {
+static void cmd_prc (RCore *core, const ut8 *block, int len) {
```
Comment (ret2libc) [inline libr/core/cmd_print.c:879]: wrong

File: libr/core/cmd_print.c [hunk 16]
```diff
@@ -642,18 +892,17 @@ static void cmd_prc(RCore *core, const ut8* block, int len) {
 	}
 	for (i = 0; i < len; i += cols) {
 		if (show_section) {
-			const char * name = r_core_get_section_name (core, core->offset + i);
-			r_cons_printf ("%20s ", name? name: "");
+			const char *name = r_core_get_section_name (core, core->offset + i);
+			r_cons_printf ("%20s ", name ? name : "");
```
Comment (ret2libc) [inline libr/core/cmd_print.c:896]: wrong

**Review by ret2libc — CHANGES_REQUESTED** [review]
I guess you are using a script (maybe even one of our own) to format the file, but there a lot of wrong things. I'd suggest to just fix the 3 LGTM warnings and keep the format of the file as-is.

**Comment by 08A** [comment]
@ret2libc i hate ClangFormat, i will fix this

**Comment by ret2libc** [comment]
> @ret2libc i hate ClangFormat, i will fix this

Eheh :D yeah, thanks!

File: libr/util/format.c (hunk 7 above)
Comment (ret2libc) [reply libr/util/format.c:509]: We don't use those things AFAIK. Just check `if (timestr) { .... ; return from function }`

Comment (ret2libc) [reply libr/util/format.c:509]: Where is `26` coming from? Better to have a macro for it or something, instead of relying on an hardcoded number.

Comment (08A) [reply libr/util/format.c:509]: man ascitime
```
      The asctime() function converts the broken-down time value tm  into  a
       null-terminated  string  with  the same format as ctime().  The return
       value points to a statically allocated string which might be overwrit‐
       ten  by  subsequent  calls to any of the date and time functions.  The
       asctime_r() function does the same, but stores the string in  a  user-
       supplied buffer which should have room for at least 26 bytes.
```

Comment (ret2libc) [reply libr/util/format.c:509]: Oh I see :/ Maybe define a macro in this file (e.g. `ASCTIME_BUF_MINLEN` or something like that).

File: libr/util/format.c [hunk 17]
```diff
@@ -506,7 +508,11 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
 	if (MUSTSET) {
 		p->cb_printf ("wv4 %s @ 0x%08"PFMT64x"\n", setval, seeki+((elem>=0)?elem*4:0));
 	} else if (MUSTSEE) {
-		char *timestr = strdup(asctime (gmtime_r ((time_t*)&addr, &timestruct)));
+		char *timestr = malloc (ASCTIME_BUF_MINLEN);
+		if (!timestr) {
+			return;
+        }
```
Comment (ret2libc) [inline libr/util/format.c:514]: wrong tab/spaces.

**Review by ret2libc — APPROVED** [review]
Please fix the tab/space issue, then it LGTM

**Comment by ret2libc** [comment]
@trufae could you please review again so we can merge? Thanks

**Review by XVilka — APPROVED** [review]

File: libr/util/format.c [hunk 18]
```diff
@@ -542,16 +547,19 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
 		}
 		free (timestr);
 	} else if (MUSTSEEJSON || MUSTSEESTRUCT) {
-		char *timestr = strdup (asctime (gmtime_r ((time_t*)&addr, &timestruct)));
+		char *timestr = malloc (ASCTIME_BUF_MINLEN);
+		if (!timestr) {
+			return;
+		}
+		asctime_r (gmtime_r ((time_t*)&addr, &timestruct), timestr);
 		*(timestr+24) = '\0';
```
Comment (radare) [inline libr/util/format.c:555]: space around +

File: libr/util/format.c [hunk 19]
```diff
@@ -542,16 +547,19 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
...
 		if (size==-1) {
 			p->cb_printf ("\"%s\"", timestr);
 		} else {
 			p->cb_printf ("[ ");
 			while (size--) {
 				updateAddr (buf + i, size - i, endian, &addr, NULL);
-				free (timestr);
-				timestr = strdup (asctime (gmtime_r ((time_t*)&addr, &timestruct)));
+				asctime_r (gmtime_r ((time_t*)&addr, &timestruct), timestr);
 				*(timestr+24) = '\0';
```
Comment (radare) [inline libr/util/format.c:563]: same

**Review by radare — APPROVED** [review]

File: libr/core/cmd_print.c [hunk 20]
```diff
@@ -4475,12 +4475,13 @@ static void r_core_disasm_table(RCore * core, int l, const char *input) {
 
 static int cmd_print(void *data, const char *input) {
 	RCore *core = (RCore *) data;
-	int i, l, len, ret;
+	ssize_t l;
```
Comment (ret2libc) [inline libr/core/cmd_print.c:4478]: ssize_t is not defined on windows, I think. Use st64 or something like that.

**Review by ret2libc — CHANGES_REQUESTED** [review]

File: libr/util/format.c [hunk 21]
```diff
@@ -542,16 +547,23 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
 		}
 		free (timestr);
 	} else if (MUSTSEEJSON || MUSTSEESTRUCT) {
-		char *timestr = strdup (asctime (gmtime_r ((time_t*)&addr, &timestruct)));
+
```
Comment (ret2libc) [inline libr/util/format.c:550]: why did you add these random newlines in the last commit?

Comment (08A) [reply libr/util/format.c:550]: @radare https://github.com/radareorg/radare2/pull/17465#discussion_r483578503

Comment (08A) [reply libr/util/format.c:550]: @radare https://github.com/radareorg/radare2/pull/17465#discussion_r483578503

Comment (08A) [reply libr/util/format.c:550]: @radare https://github.com/radareorg/radare2/pull/17465#discussion_r483578503

Comment (08A) [reply libr/util/format.c:550]: @radare https://github.com/radareorg/radare2/pull/17465#discussion_r483578503

Comment (08A) [reply libr/util/format.c:550]: @radare https://github.com/radareorg/radare2/pull/17465#discussion_r483578503

Comment (08A) [reply libr/util/format.c:550]: @radare https://github.com/radareorg/radare2/pull/17465#discussion_r483578503

Comment (08A) [reply libr/util/format.c:550]: Oh my god, the web application seems to have bugged, sorry for the ping

Comment (ret2libc) [inline libr/util/format.c:550]:
```suggestion
```

File: libr/util/format.c [hunk 22]
```diff
@@ -542,16 +547,23 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
...
+		char *timestr = malloc (ASCTIME_BUF_MINLEN);
+		if (!timestr) {
+			return;
+		}
+		asctime_r (gmtime_r ((time_t*)&addr, &timestruct), timestr);
+
```
Comment (ret2libc) [inline libr/util/format.c:556]:
```suggestion
```

File: libr/util/format.c [hunk 23]
```diff
@@ -542,16 +547,23 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
...
 		*(timestr+24) = '\0';
 		if (size==-1) {
 			p->cb_printf ("\"%s\"", timestr);
 		} else {
 			p->cb_printf ("[ ");
 			while (size--) {
 				updateAddr (buf + i, size - i, endian, &addr, NULL);
-				free (timestr);
-				timestr = strdup (asctime (gmtime_r ((time_t*)&addr, &timestruct)));
+
```
Comment (ret2libc) [inline libr/util/format.c:564]:
```suggestion
```

File: libr/util/format.c [hunk 24]
```diff
@@ -542,16 +547,23 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
...
+				asctime_r (gmtime_r ((time_t*)&addr, &timestruct), timestr);
+
```
Comment (ret2libc) [inline libr/util/format.c:566]:
```suggestion
```

**Review by ret2libc — APPROVED** [review]

**Comment by XVilka** [comment]
MSVC red:
```
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\binr\rax2\ea90354@@rax2@exe.vcxproj]
.\rax2.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\binr\rax2\ea90354@@rax2@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\binr\rafind2\921be55@@rafind2@exe.vcxproj]
.\rafind2.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\binr\rafind2\921be55@@rafind2@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\binr\rasign2\6c64c67@@rasign2@exe.vcxproj]
.\rasign2.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\binr\rasign2\6c64c67@@rasign2@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\binr\radiff2\c6fbfda@@radiff2@exe.vcxproj]
.\radiff2.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\binr\radiff2\c6fbfda@@radiff2@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\binr\r2agent\a05c824@@r2agent@exe.vcxproj]
.\r2agent.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\binr\r2agent\a05c824@@r2agent@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\binr\ragg2\749824c@@ragg2@exe.vcxproj]
.\ragg2.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\binr\ragg2\749824c@@ragg2@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\binr\radare2\cb15f3c@@radare2@exe.vcxproj]
.\radare2.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\binr\radare2\cb15f3c@@radare2@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\binr\rabin2\1154f2b@@rabin2@exe.vcxproj]
.\rabin2.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\binr\rabin2\1154f2b@@rabin2@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\binr\rasm2\198bb2c@@rasm2@exe.vcxproj]
.\rasm2.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\binr\rasm2\198bb2c@@rasm2@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\binr\rarun2\c604330@@rarun2@exe.vcxproj]
.\rarun2.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\binr\rarun2\c604330@@rarun2@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\binr\rahash2\7edda36@@rahash2@exe.vcxproj]
.\rahash2.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\binr\rahash2\7edda36@@rahash2@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\test\unit\d5b50c5@@test_cmd@exe.vcxproj]
.\test_cmd.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\test\unit\d5b50c5@@test_cmd@exe.vcxproj]
libr_util.a(format.c.obj) : error LNK2019: unresolved external symbol asctime_r referenced in function r_print_format_time [C:\projects\radare2\build\test\unit\d5b50c5@@test_anal_class_graph@exe.vcxproj]
.\test_anal_class_graph.exe : fatal error LNK1120: 1 unresolved externals [C:\projects\radare2\build\test\unit\d5b50c5@@test_anal_class_graph@exe.vcxproj]
    544 Warning(s)
    26 Error(s)
```

File: libr/util/format.c [hunk 25]
```diff
@@ -506,7 +508,11 @@ static void r_print_format_time(const RPrint* p, int endian, int mode,
 	if (MUSTSET) {
 		p->cb_printf ("wv4 %s @ 0x%08"PFMT64x"\n", setval, seeki+((elem>=0)?elem*4:0));
 	} else if (MUSTSEE) {
-		char *timestr = strdup(asctime (gmtime_r ((time_t*)&addr, &timestruct)));
+		char *timestr = malloc (ASCTIME_BUF_MINLEN);
+		if (!timestr) {
+			return;
+		}
+		asctime_r (gmtime_r ((time_t*)&addr, &timestruct), timestr);
```
Comment (trufae) [inline libr/util/format.c:515]: dont use asctime_r. its not portable. use the old implementation when on __WINDOWS__

this will fix the CI

Comment (trufae) [reply libr/util/format.c:515]: oh wait. as long as this same code is called several times it will be better to exposae this API in util/time.c and use it in every call instead of having to ifdef for windows all the time

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/util/format.c (hunk 25 above)
Comment (08A) [reply libr/util/format.c:515]: @trufae Okay, I'll do it tomorrow. Sorry for the wait.

**Comment by 08A** [comment]
The new function is `r_asctime_r(struct time *tm, char *buf, size_t size)`

The size is useless for linux.

File: libr/core/cmd_print.c [hunk 26]
```diff
@@ -4527,7 +4528,7 @@ static int cmd_print(void *data, const char *input) {
 		}
 		if (i && l > i) {
 			eprintf ("This block size is too big (0x%"PFMT64x
-				" < 0x%x). Did you mean 'p%c @ %s' instead?\n",
+				" < 0x%llx). Did you mean 'p%c @ %s' instead?\n",
```
Comment (ret2libc) [inline libr/core/cmd_print.c:4531]:
```suggestion
				" < 0x%" PFMT64x). Did you mean 'p%c @ %s' instead?\n",
```

File: libr/core/cmd_print.c [hunk 27]
```diff
@@ -5417,7 +5418,7 @@ l = use_blocksize;
 						core->num->value = r_core_print_disasm (core->print,
 							core, addr, block1, addrbytes * l, l, 0, 1, formatted_json, NULL, NULL);
 					} else {
-						eprintf ("Cannot allocate %d byte(s)\n", addrbytes * l);
+						eprintf ("Cannot allocate %lld byte(s)\n", addrbytes * l);
```
Comment (ret2libc) [inline libr/core/cmd_print.c:5421]:
```suggestion
						eprintf ("Cannot allocate %" PFMT64d " byte(s)\n", addrbytes * l);
```

**Review by ret2libc — CHANGES_REQUESTED** [review]

File: libr/core/cmd_print.c (hunk 27 above)
Comment (08A) [reply libr/core/cmd_print.c:5421]: @ret2libc Can you push those modifications. I am not on my personnal computer.

**Comment by 08A** [comment]
Ty

File: libr/core/cmd_print.c [hunk 28]
```diff
@@ -4527,7 +4528,7 @@ static int cmd_print(void *data, const char *input) {
 		}
 		if (i && l > i) {
 			eprintf ("This block size is too big (0x%"PFMT64x
-				" < 0x%x). Did you mean 'p%c @ %s' instead?\n",
+				" < 0x%" PFMT64x). Did you mean 'p%c @ %s' instead?\n",
```
Comment (ret2libc) [inline libr/core/cmd_print.c:4531]:
```suggestion
				" < 0x%" PFMT64x "). Did you mean 'p%c @ %s' instead?\n",
```

File: libr/util/time.c [hunk 29]
```diff
@@ -197,3 +197,16 @@ R_API const char *r_time_to_string (ut64 ts) {
 	l = ts >> 20;
 	return r_time_stamp_to_str (l);
 }
+
+R_API char *r_asctime_r(const struct tm *tm, char *buf, size_t size) {
+	if (size < ASCTIME_BUF_MINLEN) {
+		size = ASCTIME_BUF_MINLEN;
+	}
```
Comment (08A) [inline libr/util/time.c:204]: Line 202, 203, 204 should be deleted. I made a mistake while reading the documentation.

Comment (ret2libc) [inline libr/util/time.c:204]:
```suggestion
```

File: libr/util/time.c [hunk 30]
```diff
@@ -197,3 +197,16 @@ R_API const char *r_time_to_string (ut64 ts) {
...
+
```
Comment (ret2libc) [inline libr/util/time.c:205]:
```suggestion
```

File: libr/util/time.c [hunk 31]
```diff
@@ -197,3 +197,12 @@ R_API const char *r_time_to_string (ut64 ts) {
 	l = ts >> 20;
 	return r_time_stamp_to_str (l);
 }
+
+R_API char *r_asctime_r(const struct tm *tm, char *buf, size_t size) {
+#if __WINDOWS__
+	errno_t err = asctime_s (buf, size, tm);
+	return err ? NULL : buf;
```
Comment (ret2libc) [inline libr/util/time.c:204]:
```suggestion
	return err? NULL: buf;
```

**Review by ret2libc — APPROVED** [review]

**Review by trufae — APPROVED** [review]
